### PR TITLE
Parser: make an error on unbalanced end brace '}' in macro body

### LINKED ISSUE #5417: Macro works when it shouldn't when macro ends with just one }
Here is a minimal test case: https://play.crystal-lang.org/#/r/39kv

```crystal
macro test
  {% foo = "bar" }
  {% p foo %}
end
  
test
```

This works, but should probably fail because it is missing a %

### PR DESCRIPTION
**Pull request by MakeNowJust**
Fixed #5417

Thank you @paulcsmith for reporting this.

### CODE REVIEW

**Comment by paulcsmith** [comment]
Thanks for fixing it :D

File: spec/compiler/parser/parser_spec.cr [hunk 1]
```diff
@@ -775,6 +775,7 @@ describe "Parser" do
   it_parses "macro foo;bar{% for x in y %}\\  \n   body{% end %}\\   baz;end", Macro.new("foo", [] of Arg, Expressions.from(["bar".macro_literal, MacroFor.new(["x".var], "y".var, "body".macro_literal), "baz;".macro_literal] of ASTNode))
   it_parses "macro foo; 1 + 2 {{foo}}\\ 3 + 4; end", Macro.new("foo", [] of Arg, Expressions.from([" 1 + 2 ".macro_literal, MacroExpression.new("foo".var), "3 + 4; ".macro_literal] of ASTNode))
 
+  assert_syntax_error "macro foo; {% foo = 1 }; end"
```
Comment (RX14) [inline spec/compiler/parser/parser_spec.cr]: Maybe a check for `{{ foo = 1 }` too?

Comment (RX14) [reply spec/compiler/parser/parser_spec.cr]: Please implement this.

Comment (Sija) [reply spec/compiler/parser/parser_spec.cr]: @MakeNowJust 🏓

**Pull request merged**